nmigen/lib/roundrobin.py

```diff
@@ -0,0 +1,53 @@
+from .. import *
+
+
+__all__ = ["RoundRobin"]
+
+
+class RoundRobin(Elaboratable):
+    """Round-robin scheduler.
+
+    For a given set of requests, the round-robin scheduler will
+    grant one request.
+
+    Parameters
+    ----------
+    width : int
+        Number of requests.
+
+    Attributes
+    ----------
+    requests : Signal(width), in
+        Set of requests.
+    grant : Signal(range(width)), out
+        Number of the granted request. Indeterminate if `valid` is deasserted.
```

Comment:
I gave it more thought and realized that we can't change the current behavior without impacting fairness. I.e. consider a case where either two requests are active simultaneously, or none at all. If `grant` would change in any way when there are no active requests, the scheduler won't act as round-robin anymore--e.g. if it got reset to zero, it would always select the first request in this scenario.

So we can safely say that:
```suggestion
        Number of the granted request. Does not change if there are no active requests.
```